### ISSUE TITLE
Autocompletion of dollar-sign-prefixed variables does not work

### ISSUE THREAD

**Issue by jonathandann** [fish-shell/fish-shell#2907]
Autocompletion suggests $-prefixed variables, then fails to complete them when hitting `Tab`. Hitting `->` works.

### Reproduction Steps:

1. `cd $HOME`
2. `cd ~/Desktop`
3. `cd $<tab>`

### Expected behavior:

On step 3 `$HOME` is suggested, hitting `Tab` should complete the suggestion

### Observed behavior:

```
cd $$* is not supported. In fish, please use $argv.
- (line 1): begin; printf "%s\n" $*/ 
                                  ^
from sourcing file -
	called on line 60 of file /usr/local/Cellar/fish/2.2.0/share/fish/functions/eval.fish

in function 'eval'
	called on line 45 of file /usr/local/Cellar/fish/2.2.0/share/fish/functions/__fish_complete_cd.fish

in function '__fish_complete_cd'
	called on standard input

in command substitution
	called on standard input

source: Error while reading file '-'
```

### Additional information:

---
Fish version: fish, version 2.2.0

Operating system: OS X 10.11.4 (15E65)

Terminal or terminal emulator: OS X  Terminal

**Comment by faho** [comment]
This is already fixed in 6cb48c63800604e40ba759763ab410beceb74b89 (and a re-written cd completion).

**Comment by jonathandann** [comment]
Thank you!
